xbmc/utils/DRMHelpers.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ *  Copyright (C) 2005-2021 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DRMHelpers.h"
+
+#include "utils/StringUtils.h"
+
+#include <sstream>
+
+#include <drm_fourcc.h>
+#include <xf86drm.h>
+
+namespace DRMHELPERS
+{
+
+std::string FourCCToString(uint32_t fourcc)
+{
+  std::stringstream ss;
+  ss << static_cast<char>((fourcc & 0x000000FF));
+  ss << static_cast<char>((fourcc & 0x0000FF00) >> 8);
+  ss << static_cast<char>((fourcc & 0x00FF0000) >> 16);
+  ss << static_cast<char>((fourcc & 0xFF000000) >> 24);
+
+  return ss.str();
+}
+
+std::string ModifierToString(uint64_t modifier)
+{
+#if defined(HAVE_DRM_MODIFIER_NAME)
+  std::string modifierName(drmGetFormatModifierName(modifier));
+  std::string vendorName(drmGetFormatModifierVendor(modifier));
+
+  if (modifierName.empty())
+  {
+    if (!vendorName.empty())
+    {
+      return vendorName + "_UNKNOWN_MODIFIER";
+    }
+    else
+    {
+      return "UNKNOWN_VENDOR_UNKNOWN_MODIFIER";
+    }
+  }
```

Comment:
untested, but maybe cleaner
```suggestion
    if (modifierName.empty())
      modifierName = "UNKNOWN_MODIFIER";

  if (vendorName.empty())
      vendorName = "UNKNOWN_VENDOR";

```